### PR TITLE
Improve linear interpolation

### LINKED ISSUE #182: Linear interpolation consumes too much memory
The following code

```elixir
Application.ensure_all_started(:exla)
x = Nx.iota({1000000})
y = Nx.iota({1000000})
model = Scholar.Interpolation.Linear.fit(x, y)
EXLA.jit_apply(&Scholar.Interpolation.Linear.predict(&1, &2), [model, x])
```
gives error
```
** (RuntimeError) Out of memory allocating 1031249000016 bytes.
    (exla 0.6.0) lib/exla/executable.ex:56: EXLA.Executable.unwrap!/1
    (exla 0.6.0) lib/exla/executable.ex:19: EXLA.Executable.run/3
    (exla 0.6.0) lib/exla/defn.ex:346: EXLA.Defn.maybe_outfeed/7
    (stdlib 4.2) timer.erl:235: :timer.tc/1
    (exla 0.6.0) lib/exla/defn.ex:283: anonymous fn/7 in EXLA.Defn.__compile__/4
    (nx 0.7.0-dev) lib/nx/defn.ex:433: Nx.Defn.do_jit_apply/3
```
1TB or RAM is for linear interpolation of 1M points, pretty much.  This needs to run smoothly for this number of points because MDS for n samples calculates Linear interpolation for n ^2 samples.

### PR DESCRIPTION
**Pull request by msluszniak**
Closes #182

### CODE REVIEW

**Review by josevalim — APPROVED** [review]

**Comment by msluszniak** [comment]
I'm wondering why in the second check format is not correct. I set versions using asdf to: elixir 1.15.4 and OTP 25 and the formater doesn't change anything in this file

File: lib/scholar/interpolation/linear.ex [hunk 1]
```diff
@@ -16,12 +16,28 @@ defmodule Scholar.Interpolation.Linear do
...
+    ],
+    right: [
+      type: {:or, [:float, :integer]},
+      doc:
+        "Value to return for values in `target_x` greater that the greatest value in training set"
+    ]
+  ]
+
+  @opts_schema NimbleOptions.new!(opts_schema)
+
```
Comment (josevalim) [inline lib/scholar/interpolation/linear.ex:40]:
```suggestion

```

**Comment by josevalim** [comment]
Trailing white space :)

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**